### DISCUSSION TITLE
GenericBatchingBatcher does not apply command changes to the _currentBatch.CurrentStatement

### DISCUSSION THREAD

**Discussion by garashm2k** [nhibernate/nhibernate-core#2671] [Q&A]
We used   NHibernateUtil.String.NullSafeSet(cmd, value, index, session); to change command data run-time but when it commits the transaction, the change dose not propagated to _currentBatch.CurrentStatement

we did this change refer to the following article
https://brosteins.com/2014/07/25/sql-server-database-encryption-with-nhibernate-series-part-4-of-5/

**Comment by bahusoid** [comment]
Is it https://github.com/nhibernate/nhibernate-core/issues/2278? Can you check if your issue still exists on [master](https://github.com/nhibernate/nhibernate-core#nightly-development-builds)?

**Comment by garashm2k** [comment]
I did,it is not the same .I find the problem by cloning master 5 days ago

**Comment by maca88** [comment]
This is "by design", `CurrentStatement` is only used to populate parameters. Currently you have the following options:
- Use `NonBatchingBatcherFactory` instead
- Use a v5.3 or higher NHibernate (which has #2278 fixed), register a custom `IInterceptor` and try implement the logic inside `OnPrepareStatement` method.  [Here](https://github.com/nhibernate/nhibernate-core/pull/2285/files#diff-05c7247bc825457f970143e93a3fae0eR171) you can find an example for it.
- Use a [custom sql insert/update](https://nhibernate.info/doc/nh/en/index.html#querysql-cud)

**Comment by gtguybrush** [comment]
Hi, I too am interested in that approach but first solution is not clear: how can i use NonBatchingBatcherFactory?

in the code there are only reference to NHibernateUtil.String.NullSafeSet(cmd, value, index);

**Reply by fredericDelaporte** [reply]
By default, the batcher is chosen by the the NHibernate driver you are using (which can be chosen by the dialect), provided you have a batchSize greater than 1 (otherwise it will be the NonBatchingBatcherFactory anyway). These things are configurable, see [`adonet.factory_class` and `adonet.batch_size`](https://nhibernate.info/doc/nhibernate-reference/session-configuration.html#configuration-hibernatejdbc).

**Comment by maca88** [comment]
> how can i use NonBatchingBatcherFactory?

by setting the xml configuration property `adonet.factory_class` to `NHibernate.AdoNet.NonBatchingBatcherFactory, NHibernate` or by code:
```
var cfg = new NHibernate.Cfg.Configuration();
cfg.SetProperty(Environment.BatchStrategy, typeof(NonBatchingBatcherFactory).AssemblyQualifiedName);
```
